ReqQueueHandler.js

```diff
@@ -338,21 +357,245 @@ var SetNextProcessingItem = function (logKey, queueId, processingHashId, current
     });
     //});
 };
+*/
+
+var SetNextProcessingItem = function (logKey, queueId, processingHashId, currentSession, callback) {
+
+    var redLokKey = util.format('lock:%s:%s', processingHashId, queueId);
 
-var GetRejectedQueueId = function(queueId){
+    redisHandler.RLock.lock(redLokKey, 500).then(function (lock) {
+
+        var queueExecuteCount = util.format('ExecCount:%s', queueId);
+        redisHandler.GetObj(logKey, queueExecuteCount, function (err, exeCountStr) {
+
+            var exeCount = (err || !exeCountStr) ? 1 : parseInt(exeCountStr);
+
+            var hscanPattern = util.format('%s:*', queueId);
+            redisHandler.HScanHash(logKey, processingHashId, hscanPattern, function (processingHashDetail) {
+
+                if (processingHashDetail) {
```

Comment:
method is too long because of internal functions
some of the functions are just wrappers for redis methods
remove unnecessary internal functions to improve the readability